### PR TITLE
Use remove to cleanup canceled tasks.

### PR DESCRIPTION
**Pull request by boaks**
Fix issue #681

Signed-off-by: Achim Kraus <achim.kraus@bosch-si.com>

### CODE REVIEW

File: californium-core/src/main/java/org/eclipse/californium/core/CoapObserveRelation.java [hunk 1]
```diff
@@ -59,7 +60,7 @@
 	private static final Logger LOGGER = LoggerFactory.getLogger(CoapObserveRelation.class.getCanonicalName());
 
 	/** A executor service to schedule re-registrations */
-	private static final ScheduledExecutorService scheduler = Executors.newSingleThreadScheduledExecutor(//
+	private static final ScheduledThreadPoolExecutor scheduler = new ScheduledThreadPoolExecutor(1, //
```
[the remark this answers is not in the document]
Comment (boaks) [reply californium-core/src/main/java/org/eclipse/californium/core/CoapObserveRelation.java]: Executors create a `ScheduledThreadPoolExecutor`, but it's hidden and could not be accessed. 
But to remove, you need that access to at least `ThreadPoolExecutor`, and to schedule, you need access to a `ScheduledExecutor`, so I decided to go for `ScheduledThreadPoolExecutor` directly.

File: californium-core/src/main/java/org/eclipse/californium/core/CoapObserveRelation.java [hunk 2]
```diff
@@ -275,21 +287,20 @@ protected boolean onResponse(CoapResponse response) {
...
-				@Override
-				public void run() {
-					reregister();
-				}
-			}, timeout, TimeUnit.MILLISECONDS);
+			ScheduledFuture<?> f = scheduler.schedule(reregister, timeout, TimeUnit.MILLISECONDS);
 			setReregistrationHandle(f);
 		}
 	}
+
```
[the remark this answers is not in the document]
Comment (boaks) [reply californium-core/src/main/java/org/eclipse/californium/core/CoapObserveRelation.java]: no :-)

File: californium-core/src/main/java/org/eclipse/californium/core/CoapObserveRelation.java [hunk 3]
```diff
@@ -90,6 +91,17 @@
...
+	/**
+	 * Task to schedule {@link CoapObserveRelation#reregister()}.
+	 */
+	private final Runnable reregister = new Runnable() {
+
+		@Override
+		public void run() {
+			reregister();
+		}
+	};
```
[the remark this answers is not in the document]
Comment (boaks) [reply californium-core/src/main/java/org/eclipse/californium/core/CoapObserveRelation.java]: Yes.

File: californium-core/src/main/java/org/eclipse/californium/core/CoapObserveRelation.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (sbernard31) [reply californium-core/src/main/java/org/eclipse/californium/core/CoapObserveRelation.java]: Ok I get it.

**Pull request merged**